app/models/concerns/patient_reportable.rb

```diff
@@ -9,19 +9,27 @@ module PatientReportable
     scope :with_hypertension, -> { joins(:medical_history).merge(MedicalHistory.hypertension_yes).distinct }
     scope :excluding_dead, -> { where.not(status: :dead) }
 
-    scope :ltfu_as_of, ->(date) do
+    scope :ltfu_as_of, ->(time) do
+      bp_start = reporting_timestamp((time - LTFU_TIME).end_of_month)
+      bp_end = reporting_timestamp(time)
+      patient_registration_cutoff = (time - LTFU_TIME).end_of_month
+
       joins("LEFT OUTER JOIN latest_blood_pressures_per_patient_per_months
              ON patients.id = latest_blood_pressures_per_patient_per_months.patient_id
-             AND #{sanitize_sql(["bp_recorded_at > ? AND bp_recorded_at < ?", (date.to_date - LTFU_TIME).end_of_month, date])}")
-        .where("bp_recorded_at IS NULL AND patients.recorded_at < ?", (date.to_date - LTFU_TIME).end_of_month)
+             AND #{sanitize_sql(["bp_recorded_at > ? AND bp_recorded_at < ?", bp_start, bp_end])}")
+        .where("bp_recorded_at IS NULL AND patients.recorded_at <= ?", patient_registration_cutoff)
         .distinct(:patient_id)
     end
 
-    scope :not_ltfu_as_of, ->(date) do
+    scope :not_ltfu_as_of, ->(time) do
+      bp_start = reporting_timestamp((time - LTFU_TIME).end_of_month)
```

Comment:
A better name for this method could be `legacy_matview_timestamp`, to be very clear when this should be used.